bootloader/prebootloader/src/rtl872x/part1/sleep_handler.cpp

```diff
@@ -327,13 +331,22 @@ void sleepProcess(void) {
             auto config = sleepConfigShadow.config();
             if (config->mode == HAL_SLEEP_MODE_HIBERNATE) {
                 SysTick->CTRL &= ~SysTick_CTRL_ENABLE_Msk;
-                taskENTER_CRITICAL();
+                if (xTaskGetSchedulerState() != taskSCHEDULER_NOT_STARTED) {
```

Comment:
Just wondering why this matters. Is it the root cause of the sleep duration being incorrect?

Reply:
In non-RTOS mode (when WiFi firmware is not initialized) this will disable interrupts and we'll no longer wake up KM0 on IPC interrupts. So 5 second 'sleep' was actually just a 5 second timeout waiting to get an IPC response.